torch/csrc/aot_inductor/c/model_interface.h

```diff
@@ -59,9 +51,9 @@ AOTInductorError AOTInductorModelContainerDelete(
 // Runs the inference.
 AOTInductorError AOTInductorModelContainerRun(
     AOTInductorModelContainerHandle container_handle,
-    const AOTInductorTensorHandle inputs_handle,
+    AtenTensorHandle* inputs_handle,
     size_t num_inputs,
-    AOTInductorTensorHandle outputs_handle,
+    AtenTensorHandle* outputs_handle,
     size_t num_outputs,
     AOTInductorStreamHandle stream_handle);
```

Comment:
Maybe we need to phone in @chenyang78, but I don't understand model_interface.h at all. Naively, I expected this to be the C ABI which you can access model.so from the runtime. However, I would NOT expect it to be a plain list of functions, because I am dlopen'ing the model.so and I am getting a pile of function pointers that I would probably assign into a struct to dynamic dispatch into. So I'd expect a bunch of function pointer signatures, not... this. What the heck is this?

It is pointless for the header file that predictor service links against to be ABI compatible if the implementation is linked into predictor, because you can assume that this will always be compiled together, so you can make sure they're synchronized in lockstep. Am I misunderstanding the situation here?

Reply:
@ezyang The function pointers are declared in the internal runtime header file, AOTInductorModelImpl.h. It is included in the source file (AOTInductorModelImpl.cpp) where we dlopen the model.so. This was how the AIT was originally designed so that I just re-used a similar mechanism. I think one difference is that AIT also open-sources the corresponding runtime part of the code (https://github.com/facebookincubator/AITemplate/blob/main/fx2ait/fx2ait/csrc/AITModelImpl.h), so it's probably less confused to some extent.

Reply:
Yes, please open source this code lol

Reply:
@chenyang78 Then, in that case, model_interface.h doesn't actually have any C ABI requirement, and it shouldn't be placed in the c/ folder

Reply:
@ezyang I think we have two sets of C interfaces, one declared in aten_shim.h and one declared in model_interface.h. The former is supposed to be used internally in the aot_inductor_model.so for mitigating ABI compatibility issue.  The latter is used by the external user code that loads aot_inductor_model.so. 

Is the c/ folder meant for holding only internal ABI-related files? If Yes, we should probably move model_interface.h to somewhere else. On the other hand, if the c/ folder is the place for all C interface files, it might be fine to leave model_interface.h there?